### PR TITLE
Redefine reduceLeft/Right in terms of foldLeft/Right, remove reduceLe…

### PR DESCRIPTION
**Pull request by esbenbjerre**
I just realized that the reduceLeft/RightHelper is just a slightly specialized fold and thus redundant.

### CODE REVIEW

File: main/src/library/RedBlackTree.flix [hunk 1]
```diff
@@ -178,17 +178,16 @@ namespace RedBlackTree {
...
     ///
     /// That is, the result is of the form: `Some(f(...f(f(k1, v1, k2, v2), k3, v3)..., kn, vn))`
     ///
     /// Returns `None` if `tree` is the empty tree.
     ///
     pub def reduceLeft(f: (k, v, k, v) -> (k, v) & e, tree: RedBlackTree[k, v]): Option[(k, v)] & e =
-        reduceLeftHelper((x: Option[(k, v)], y: (k, v)) -> match x {
-            case Some((k, v)) => Some(f(k, v, fst(y), snd(y)))
-            case None => Some(y)
+        foldLeft((x: Option[(k, v)], k1: k, v1: v) -> match x {
```
Comment (magnus-madsen) [inline main/src/library/RedBlackTree.flix:188]: Did you put the type annotation to help you? Or did it simply not compile without it?

**Comment by magnus-madsen** [comment]
Looks good. I will merge, but I had one question.

File: main/src/library/RedBlackTree.flix (hunk 1 above)
Comment (esbenbjerre) [reply main/src/library/RedBlackTree.flix]: No idea, an old habit from Scala maybe.
I have tested it without and removed it.

**Comment by magnus-madsen** [comment]
Thanks!

**Pull request merged**